src/shaders/fill_extrusion.vertex.glsl

```diff
@@ -155,25 +166,40 @@ void main() {
     y_ground += y_ground * 5.0 / max(3.0, top_height);
 #endif
     v_ao = vec3(mix(concave, -concave, start), y_ground, h - ele);
-    directional *= (1.0 + 0.05 * (1.0 - top_up_ny.y) * u_ao[0]); // compensate sides faux ao shading contribution
+    NdotL *= (1.0 + 0.05 * (1.0 - top_up_ny.y) * u_ao[0]); // compensate sides faux ao shading contribution
 
 #ifdef PROJECTION_GLOBE_VIEW
     top_height += u_height_lift;
 #endif
     gl_Position.z -= (0.0000006 * (min(top_height, 500.) + 2.0 * min(base, 500.0) + 60.0 * concave + 3.0 * start)) * gl_Position.w;
 #endif
 
-    // Assign final color based on surface + ambient light color, diffuse light directional, and light color
+    vec3 indirect = vec3(0.0);
+    vec3 direct = vec3(0.0);
+#ifdef LIGHTING_3D_MODE
+    indirect = u_lighting_ambient_color * color.rgb;
+    direct = u_lighting_directional_color * color.rgb;
+    v_color.rgb += indirect + direct * NdotL;
```

Comment:
why `+=`?

Reply:
I haven't modified this bit, v_color.w is already set to 1 earlier.